fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/DistributionDisjointSet.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.base;
+
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import java.util.function.Function;
+
+// it's a disjoint-set data structure to record derived distribution cols from equalPredicate in on clause
+public class DistributionDisjointSet {
```

Comment:
get a better name? disjoint-set is a special word? it seems that it has no relation with equals?

Reply:
Yes, disjoint-set means a special data structure.